plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -340,6 +345,17 @@ public PostgreSqlClient(
                         .build());
     }
 
+    @VisibleForTesting
+    Optional<String> mapTypeSignature(ConnectorSession connectorSession, TypeId typeId)
+    {
+        try {
+            return Optional.of(this.toWriteMapping(connectorSession, typeManager.getType(typeId)).getDataType());
```

Comment:
**If** that was correct, we could have BaseJdbc provide such logic and it wouldn't need to be connector-specific.